Branch-SDK/src/io/branch/referral/util/CommerceEvent.java

```diff
@@ -137,6 +131,7 @@ public JSONObject getCommerceJSONObject() {
     }
 
     public List<JSONObject> getProducts() {
+        if (this.products == null) return null;
```

Comment:
I returned null there because @derrickstaten doesn't want empty items in the payload, and this will return an empty product item.